src/goto-symex/symex_assign.cpp

```diff
@@ -350,6 +358,92 @@ void symex_assignt::assign_from_struct(
   }
 }
 
+void symex_assignt::assign_constants(
+  const ssa_exprt &length,
+  const constant_exprt &new_length,
+  const ssa_exprt &char_array,
+  const array_exprt &new_char_array)
+{
+  // assign length of string
+  assign_symbol(
+    length,
+    expr_skeletont{},
+    new_length,
+    {});
+
+  symbol_table_baset &st_mutable =
+    const_cast<symbol_table_baset &>(ns.get_symbol_table());
+
+  array_typet new_char_array_type(
+    char_type,
+    new_length);
+
+  symbolt &aux_symbol = get_fresh_aux_symbol(
```

Comment:
:pick: give a better name than `aux_symbol`, one that would make clear that it is a char array